File: 이것이코딩테스트다/greedy/큰수의_법칙.js
const fs = require('fs')
const input = fs.readFileSync('input.txt').toString().trim().split('\n');

let [N, M, K] = input[0].split(' ').map(Number);
let array = input[1].split(' ').map(Number);

array.sort((a, b) => b - a)


// 가장 큰 수를 K번 더하기
// M이 남았다면 
// 그 다음 큰수 1번 더하기
let answer = 0

while (true) {
  for (let i = 0; i < K; i++) {
    if (M === 0) {
      break
    }
    answer += array[0]
    M -= 1
  }
  if (M === 0) {
    break
  }
  answer += array[1]
  M -= 1
}

console.log(answer)

File: 이것이코딩테스트다/greedy/큰수의_법칙.py
N, M, K = map(int, input().split())
array = list(map(int, input().split()))

array.sort()
print(array)

first = array[-1]
second = array[-2]

answer = 0

while(True):
  for i in range(K):
    if M == 0:
      break
    answer += first
    M -= 1
  if M == 0:
    break
  answer += second
  M -= 1

print(answer)
